### PR TITLE
misc fixes

### PR DESCRIPTION
**Pull request by elia**
- ~~Avoid RSpec 3.5 which drops the text mate formatter~~
- Use case/when instead of if/elsif
- Support RSpec.describe in syntax
- Detect files using autorun
- Improve described class for rails
- Use rails_helper by default for Rails apps (updated)

Let me know if you prefer separate PRs

### CODE REVIEW

File: Support/Gemfile [hunk 1]
```diff
@@ -1,3 +1,3 @@
 source "http://rubygems.org"
 
-gem "rspec", '~> 3.3'
+gem "rspec", ['~> 3.3', '< 3.5']
```
Comment (noniq) [inline Support/Gemfile]: Hm, why is this necessary? `TextMateFormatter` is now part of this bundle (see https://github.com/rspec/rspec.tmbundle/blob/master/Support/lib/rspec/mate/text_mate_formatter.rb), so the bundle should work with all RSpec versions.

File: Syntaxes/RSpec.tmLanguage [hunk 2]
```diff
@@ -92,7 +94,7 @@
 		<key>behaviour</key>
 		<dict>
 			<key>begin</key>
-			<string>^\s*([fx]?(?:describe|context|feature))\b(?=.*\bdo\s*$)</string>
+			<string>^\s*([fx]?(?:RSpec\.describe|describe|context|feature))\b(?=.*\bdo\s*$)</string>
```
Comment (noniq) [inline Syntaxes/RSpec.tmLanguage]: This won’t work for `RSpec.fdescribe` and `RSpec.xdescribe` (but it would match e.g. `fRSpec.describe`, which doesn’t make sense). 

As far as I understand, all methods mentioned in the “Defining Example Groups” section of http://rspec.info/documentation/3.5/rspec-core/RSpec/Core/ExampleGroup.html can be used at the outermost level, so the grammar should accept the `RSpec.` prefix for all of them … (I might be wrong, though – what do you think?)

File: Support/Gemfile (hunk 1 above)
Comment (elia) [inline Support/Gemfile]: I got an error related to the absence of the `textmate` formatter, and jumped immediately to the gemfile with that change. But I can't reproduce it so maybe it was something else.

(reverted)

File: Syntaxes/RSpec.tmLanguage (hunk 2 above)
Comment (elia) [inline Syntaxes/RSpec.tmLanguage]: That's right, this was a quick fix I had locally since we switched to the explicit call.
Fixed in dc77c35

Comment (noniq) [inline Syntaxes/RSpec.tmLanguage]: Thanks. Let me know if it happens again and there is a way to reproduce it!

File: Support/Gemfile (hunk 1 above)
Comment (noniq) [inline Support/Gemfile]: http://asset-5.soupcdn.com/asset/16153/4806_5d6e_520.jpeg

Comment (noniq) [inline Support/Gemfile]: Thanks! Let me know if it happens again and there is a way to reproduce it.

File: Support/lib/rspec/mate/switch_command.rb [hunk 3]
```diff
@@ -140,8 +140,9 @@ def klass(relative_path, content=nil)
       def write_and_open(path)
         FileUtils.mkdir_p(File.dirname(path))
         described = described_class_for(path, base_dir)
+        is_rails = File.directory? path.sub(%r{/spec/.*$}, '/app')
```
Comment (noniq) [inline Support/lib/rspec/mate/switch_command.rb]: I think this check may be too unspecific … are there other frameworks also using a top-level `app` directory? But right now I can’t think of a good alternative either (apart from parsing `Gemfile.lock` to see if it include `rails`) … any ideas?

File: Syntaxes/RSpec.tmLanguage (hunk 2 above)
Comment (noniq) [inline Syntaxes/RSpec.tmLanguage]: I think you fixed it for one of the other two cases, but here it is still unfixed :)

File: Syntaxes/RSpec.tmLanguage [hunk 4]
```diff
@@ -209,7 +211,7 @@
 			<key>comment</key>
 			<string>This only works with simple single / double quoted strings. But matching too liberally causes bugs (see https://github.com/rspec/rspec-tmbundle/issues/45), so better err on the safe side.</string>
 			<key>match</key>
-			<string>^\s*([fx]?(?:it|specify|scenario|describe|context|feature))\s+("[^"]*"|'[^']*')(?&lt;!do)\s*$</string>
+			<string>^\s*([fx]?(?:it|specify|scenario|RSpec\.describe|describe|context|feature))\s+("[^"]*"|'[^']*')(?&lt;!do)\s*$</string>
```
Comment (noniq) [inline Syntaxes/RSpec.tmLanguage]: Please fix this occurence, too.

File: Support/lib/rspec/mate/switch_command.rb (hunk 3 above)
Comment (elia) [inline Support/lib/rspec/mate/switch_command.rb]: I agree the whole rails thing could use some refactoring. That said if I'm not mistaken checking for `app/` is already used to detect rails in other parts of the code.
This should work most of the time, possibly other frameworks use the same pattern inside `spec/` too…

File: Syntaxes/RSpec.tmLanguage (hunk 2 above)
Comment (elia) [inline Syntaxes/RSpec.tmLanguage]: Had a second look at how this is dealing with the leading `RSpec.` and fc9d4cf should be a definite improvement. Fixes all the occurences and the symbol list.

File: Support/lib/rspec/mate/switch_command.rb (hunk 3 above)
Comment (noniq) [inline Support/lib/rspec/mate/switch_command.rb]: I’m only concerned with writing `require 'rails_helper'` if we (incorrectly) detect a Rails app. Maybe in this case we could simply check for the existence of `rails_helper.rb` and use this to decide whether to require `rails_helper` or `spec_helper`?

Comment (elia) [inline Support/lib/rspec/mate/switch_command.rb]: 👍🏻 Fine by me. Ideally I think there should be a single (reliable?) method dedicated to rails detection. 

(I'll push the change later)

**Comment by noniq** [comment]
A general note: As far as I understand the new system for code-reviewing at GitHub works better if you don’t rebase your commits during the review (because than it’s easier to track which and how the review comments have been addressed in additional comments). So please just push additional commits, we can then do a final rebase at the end before merging 😀

**Comment by elia** [comment]
@noniq is there anything else I should do for this PR?

> So please just push additional commits, we can then do a final rebase at the end before merging 😀

Yes of course sorry, still had to learn how the review system works 😃

**Comment by noniq** [comment]
Looks fine, thanks! Merged 👍
